app/views/govuk_publishing_components/components/docs/radio.yml

```diff
@@ -85,8 +85,22 @@ examples:
         text: "Use Government Gateway"
       - value: "govuk-verify"
         text: "Use GOV.UK Verify"
+  with_legend:
+    description: |
+      This adds a `legend` element containing the text supplied. 
+      
+      By default, this will render as a `h2` inside the legend. This can be changed to a different `heading` level using the `heading_level` option to `0`
```

Comment:
Wording suggestion: `Legend text is automatically wrapped inside a H2. To render the text without it,  heading_level must be set to 0.`